File: src/Tools/Dova.Tools.DefinitionGenerator/Models/ClassElementDefinitionModel.cs
namespace Dova.Tools.DefinitionGenerator.Models;

internal class ClassElementDefinitionModel
{
    public string Modifiers { get; set; }
    public string ReturnType { get; set; }
    public string Name { get; set; }
    public IReadOnlyList<ParameterDefinitionModel> ParameterModels { get; set; } = new List<ParameterDefinitionModel>();
    public IReadOnlyList<TypeParameterModel> TypeParameterModels { get; set; } = new List<TypeParameterModel>();
    public string Signature { get; set; }
    public bool IsStatic { get; set; }
    public bool HasParent { get; set; }
}

File: src/Tools/Dova.Tools.Java.ClassGenerator/src/main/java/com/github/sejoslaw/dova/ClassElementDefinitionModel.java
package com.github.sejoslaw.dova;

import java.util.ArrayList;
import java.util.Collection;

public class ClassElementDefinitionModel {
    public String Modifiers;
    public String ReturnType;
    public String Name;
    public Collection<ParameterDefinitionModel> ParameterModels = new ArrayList<>();
    public Collection<TypeParameterModel> TypeParameterModels = new ArrayList<>();
    public String Signature;
    public boolean IsStatic;
    public boolean HasParent;
}
